azkustodata/kql/query_parameters.go

```diff
@@ -1,70 +1,74 @@
 package kql
 
 import (
-	"github.com/Azure/azure-kusto-go/azkustodata/types"
+	"github.com/Azure/azure-kusto-go/azkustodata/value"
 	"github.com/google/uuid"
 	"github.com/shopspring/decimal"
 	"strings"
 	"time"
 )
 
 type Parameters struct {
-	parameters map[string]Value
+	parameters map[string]value.Kusto
 }
 
 func NewParameters() *Parameters {
-	return &Parameters{parameters: make(map[string]Value)}
+	return &Parameters{parameters: make(map[string]value.Kusto)}
 }
 
 func (q *Parameters) Count() int {
 	return len(q.parameters)
 }
-func (q *Parameters) addBase(key string, value Value) *Parameters {
+func (q *Parameters) AddValue(key string, v value.Kusto) *Parameters {
 	if RequiresQuoting(key) {
 		panic("Invalid parameter values. make sure to adhere to KQL entity name conventions and escaping rules.")
 	}
-	q.parameters[key] = value
+	q.parameters[key] = v
 	return q
 }
 
-func (q *Parameters) AddBool(key string, value bool) *Parameters {
-	return q.addBase(key, newValue(value, types.Bool))
+func (q *Parameters) AddBool(key string, v bool) *Parameters {
```

Comment:
Same question - does this support nulls?

Reply:
Same answer